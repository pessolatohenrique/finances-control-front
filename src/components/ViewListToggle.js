import { Button, Grid } from "@mui/material";
import ListIcon from "@material-ui/icons/List";
import ViewModuleIcon from "@material-ui/icons/ViewModule";

function ViewListToggle({ isTable, isList, switchFormat }) {
  return (
    <Grid container justifyContent="flex-end" gap="10px">
      <Button
        data-testid="view-table"
        variant={isTable() ? "contained" : "text"}
        onClick={() => switchFormat("table")}
      >
        <ListIcon />
      </Button>
      <Button
        data-testid="view-card"
        variant={isList() ? "contained" : "text"}
        onClick={() => switchFormat("list")}
      >
        <ViewModuleIcon />
      </Button>
    </Grid>
  );
}

export default ViewListToggle;
