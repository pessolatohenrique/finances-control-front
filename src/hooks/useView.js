import { useState } from "react";

function useView() {
  const [format, setFormat] = useState("table");

  function switchFormat(format) {
    setFormat(format);
  }

  function isTable() {
    return format === "table";
  }

  function isList() {
    return format === "list";
  }

  return [isTable, isList, switchFormat];
}

export default useView;
