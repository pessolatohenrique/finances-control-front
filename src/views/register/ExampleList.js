import React from "react";
import {
  Container,
  Card,
  CardContent,
  CardActions,
  Typography,
  Button,
  TableContainer,
  Table,
  TableHead,
  TableBody,
  TableCell,
  TableRow,
  Grid,
  List,
  ListItem,
  ListItemText,
} from "@mui/material";
import { StyledTableCell } from "../../constants/CustomStyles";
import { THEME_COLOR } from "../../constants/default_settings";
import BreadcrumbsWrapper from "../../components/BreadcrumbsWrapper";
import ViewListToggle from "../../components/ViewListToggle";
import useView from "../../hooks/useView";

function createData(name, author, pages, genre) {
  return { name, author, pages, genre };
}

const rows = [
  createData("Jogos Vorazes", "Suzanne Collins", 320, "Trilogia"),
  createData(
    "Harry Potter e a Pedra Filosofal",
    "J.K Rowling",
    220,
    "Trilogia"
  ),
  createData("Me poupe!", "Nathalia Arcuri", 200, "Finanças"),
  createData("Sherlock Homes", "Arthur Conan Doyle", 500, "Suspense"),
  createData(
    "O mistério dos sete relógios",
    "Agatha Christie",
    300,
    "Suspense"
  ),
];

function ExampleList() {
  const [isTable, isList, switchFormat] = useView();

  return (
    <>
      <Container fixed>
        <br />
        <Card>
          <CardContent>
            <BreadcrumbsWrapper childrenLabel="Livros" />
            <Typography
              gutterBottom
              variant="h5"
              component="h1"
              color={THEME_COLOR}
            >
              Livros
            </Typography>

            <ViewListToggle
              isTable={isTable}
              isList={isList}
              switchFormat={switchFormat}
            />

            {isTable() && (
              <TableContainer>
                <Table>
                  <TableHead>
                    <TableRow>
                      <StyledTableCell>Título</StyledTableCell>
                      <StyledTableCell>Autor</StyledTableCell>
                      <StyledTableCell>Gênero</StyledTableCell>
                      <StyledTableCell>Páginas</StyledTableCell>
                    </TableRow>
                  </TableHead>
                  <TableBody>
                    {rows.map((row) => {
                      return (
                        <TableRow key={row.name}>
                          <TableCell>{row.name}</TableCell>
                          <TableCell>{row.author}</TableCell>
                          <TableCell>{row.genre}</TableCell>
                          <TableCell>{row.pages}</TableCell>
                        </TableRow>
                      );
                    })}
                  </TableBody>
                </Table>
              </TableContainer>
            )}

            {isList() && (
              <Grid container spacing={2}>
                {rows.map((row) => {
                  return (
                    <Grid item xs={4} key={row.name}>
                      <Card>
                        <CardContent>
                          <Typography
                            variant="h6"
                            component="div"
                            color="text.secondary"
                          >
                            {row.name}
                          </Typography>
                          <List variant="body2">
                            <ListItem>
                              <ListItemText disableTypography>
                                <Typography variant="body2">
                                  {row.author}
                                </Typography>
                              </ListItemText>
                            </ListItem>
                            <ListItem>
                              <ListItemText disableTypography>
                                <Typography variant="body2">
                                  {row.genre}
                                </Typography>
                              </ListItemText>
                            </ListItem>
                            <ListItem>
                              <ListItemText disableTypography>
                                <Typography variant="body2">
                                  {row.pages}
                                </Typography>
                              </ListItemText>
                            </ListItem>
                          </List>
                        </CardContent>
                      </Card>
                    </Grid>
                  );
                })}
              </Grid>
            )}
          </CardContent>
          <CardActions>
            <Button size="small" href="/autores/novo">
              Adicionar
            </Button>
          </CardActions>
        </Card>
      </Container>
    </>
  );
}

export default ExampleList;
