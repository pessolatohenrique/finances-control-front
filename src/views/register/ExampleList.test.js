import { render, screen, fireEvent } from "@testing-library/react";
import { renderHook } from "@testing-library/react-hooks";
import useView from "../../hooks/useView";
import ViewListToggle from "../../components/ViewListToggle";
import ExampleList from "./ExampleList";

describe("ExampleList", () => {
  it("should generate snapshot", () => {
    const { container } = render(<ExampleList />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render screen", () => {
    render(<ExampleList />);
    expect(screen.getAllByText("Livros")).toHaveLength(2);
  });

  it("should render table when toggle", () => {
    const { container } = render(<ExampleList />);

    const tableBtn = screen.getByTestId("view-table");
    fireEvent.click(tableBtn);

    expect(container.getElementsByClassName("MuiTable-root")).toHaveLength(1);
  });

  it("should render list when toggle", () => {
    const { container } = render(<ExampleList />);

    const tableBtn = screen.getByTestId("view-card");
    fireEvent.click(tableBtn);

    expect(
      container.getElementsByClassName("MuiGrid-spacing-xs-2")
    ).toHaveLength(1);
  });
});
