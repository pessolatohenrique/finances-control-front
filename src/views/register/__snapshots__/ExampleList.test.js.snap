// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleList should generate snapshot 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed css-1ss2ei0-MuiContainer-root"
>
  <br />
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineHover css-t7xl82-MuiTypography-root-MuiLink-root"
              data-testid="bread-firstlevel"
              href="/"
            >
              Início
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            ›
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-1yr8029-MuiTypography-root"
              data-testid="bread-lastlevel"
            >
              Livros
            </p>
          </li>
        </ol>
      </nav>
      <h1
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1y15g74-MuiTypography-root"
      >
        Livros
      </h1>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-1mx4t5n-MuiGrid-root"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="view-table"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ListIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 13h2v-2H3v2zm0 4h2v-2H3v2zm0-8h2V7H3v2zm4 4h14v-2H7v2zm0 4h14v-2H7v2zM7 7v2h14V7H7z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="view-card"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ViewModuleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4 11h5V5H4v6zm0 7h5v-6H4v6zm6 0h5v-6h-5v6zm6 0h5v-6h-5v6zm-6-7h5V5h-5v6zm6-6v6h5V5h-5z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-rqglhn-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1w4f6gj-MuiTableCell-root"
                scope="col"
              >
                Título
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1w4f6gj-MuiTableCell-root"
                scope="col"
              >
                Autor
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1w4f6gj-MuiTableCell-root"
                scope="col"
              >
                Gênero
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1w4f6gj-MuiTableCell-root"
                scope="col"
              >
                Páginas
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Jogos Vorazes
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Suzanne Collins
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Trilogia
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                320
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Harry Potter e a Pedra Filosofal
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                J.K Rowling
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Trilogia
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                220
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Me poupe!
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Nathalia Arcuri
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Finanças
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                200
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Sherlock Homes
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Arthur Conan Doyle
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Suspense
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                500
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                O mistério dos sete relógios
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Agatha Christie
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Suspense
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                300
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
    >
      <a
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root css-1knaqv7-MuiButtonBase-root-MuiButton-root"
        href="/autores/novo"
        tabindex="0"
      >
        Adicionar
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</div>
`;
